pkg/runtime/runtime.go

```diff
@@ -1074,12 +1074,27 @@ func (a *DaprRuntime) initNameResolution() error {
 
 func (a *DaprRuntime) publishMessageHTTP(msg *pubsub.NewMessage) error {
 	var cloudEvent map[string]interface{}
+
 	err := a.json.Unmarshal(msg.Data, &cloudEvent)
 	if err != nil {
 		log.Debug(errors.Errorf("failed to deserialize cloudevent: %s", err))
 		return err
 	}
 
+	contentType := cloudEvent[pubsub.DataContentTypeField].(string)
+	if !contenttype.IsStringContentType(contentType) && !contenttype.IsJSONContentType(contentType) {
+		// this is a binary content type
+		cloudEvent["data_base64"] = cloudEvent[pubsub.DataField]
```

Comment:
This is a breaking change to how we handle binary data now. All SDKs would need to change and apps too. Is there a way to make this backwards compatible?

Reply:
Yes, there is a way to make it backward compatible
We can add a new field `data_base64` for binary data which can contain base64 encoded value and let `data` be as it is. 
I think most SDKs (I checked dotnet-sdk, go-sdk, php-sdk) deserializers will ignore the new `data_base64` field. 
Later, when all SDKs and app handle binary data properly by considering `data_base64` then we can remove `data` for binary data. 

The temporary impact would be that the payload size will be double for binary data. 
@artursouza Let me know if this is fine

Reply:
I'm fine with that. @artursouza?

Reply:
LGTM too.

Reply:
Now, thinking about it again. Is there a way for this to be configurable? So this behavior can be opted in.

Reply:
You mean by passing a parameter or in config. Do you have any use cases in mind for the need of configuration

Reply:
In config would be better IMO, so this is a choice for a particular pubsub config not for the whole runtime. So, the binary data can be handled in one place only.

Reply:
There are multiple places this can be done:

1. Add a metadata in component config file. However, this configuration should not be used because it specifies the configuration of the pubsub infra/broker and not the subscriber/ publisher.
For example: 
```
metadata:
     - name: handleBinaryData
       value: true/false
```
2. We can add an optional property to be returned by the app on invocation of `dapr/subscribe` endpoint. However, when we use SDKs, these endpoints are consumed by the SDKs itself [example](https://github.com/dapr/dotnet-sdk/blob/0ccddf73ffe74ab45881825d56dc12beab5cb699/src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs). So there is no benefit from adding a optional property there. 

I do agree that binary data should be handled at a single place but (IMO) it should not be configurable.

Reply:
I think this should not be configurable.. this is a spec conformance issue. I wouldn't want to give users an option to tweak unless it has a very clear value.

Reply:
So, we should make this a breaking change and fix in the SDKs. Duplicating the data is worse, IMO.

Reply:
We are close to a deadline to code freeze. So, I took the liberty to address the comments on this change. While doing so, I realized that this approach has a major problem: it can modify the payload on behalf of the publisher instead of sending the CE as-is. So, I made this change in components contrib instead: https://github.com/dapr/components-contrib/pull/674/files